pkg/endpoint/endpoint.go

```diff
@@ -77,32 +77,35 @@ var (
 	EndpointMutableOptionLibrary = option.GetEndpointMutableOptionLibrary()
 )
 
+// State is an enumeration for possible endpoint states.
+type State string
```

Comment:
`State` is not being used a receiver type to define methods so it can defined as a type alias (i.e.type State = string) as opposed to a new type. That way, we can get rid of these unnecessary type conversions below at several places.

Reply:
Hi, I'm not sure about using type aliases here.
By introducing a new type, we've gained a little bit of type safety.
I.e. passing a variable of type other than State in `SetState` ends with a compile-time error.
The explicit conversion back to string is only the opposite version of the same idea.
We don't have this safety with type alias. We could still pass a random string variable in the setter. So what's the purpose of adding an alias to string?
Also, I've not noticed using type aliases in that way in this repo.

Reply:
Agreed, that's a fair point.

>passing a variable of type other than State in SetState ends with a compile-time error.

You can still pass an arbitrary string to `SetState`, like so - `ep.SetState("foo", "bar")`
Here is a playground example - https://play.golang.org/p/3iBGrWdmikq. Compile type assertions will prevent passing a `string` type variable to the setter, which could be still worthwhile to have.

Reply:
> You can still pass an arbitrary string to SetState
True, I also mentioned that https://github.com/cilium/cilium/issues/14841#issuecomment-816275822.
Trying to prevent this would obscure the code, so I think this solution is a good compromise.